crates/ruff_linter/src/suppression.rs

```diff
@@ -0,0 +1,456 @@
+use core::fmt;
+use itertools::Itertools;
+use std::{error::Error, fmt::Formatter};
+use thiserror::Error;
+
+use ruff_python_trivia::{CommentRanges, Cursor};
+use ruff_text_size::{TextRange, TextSize};
+use smallvec::{SmallVec, smallvec};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+enum SuppressionAction {
+    Disable,
+    Enable,
+    Ignore,
```

Comment:
I'm leaning towards skipping `ignore` for now, given that we don't have consensus on a design yet